validator/client/beacon-api/duties.go

```diff
@@ -75,17 +74,20 @@ func (c *beaconApiValidatorClient) duties(ctx context.Context, in *ethpb.DutiesR
 	}
 
 	return &ethpb.ValidatorDutiesContainer{
-		CurrentEpochDuties: currentEpochDuties,
-		NextEpochDuties:    nextEpochDuties,
+		PrevDependentRoot:  currentEpochDuties.PrevDependentRoot,
+		CurrDependentRoot:  currentEpochDuties.CurrDependentRoot,
+		CurrentEpochDuties: currentEpochDuties.CurrentEpochDuties,
+		NextEpochDuties:    nextEpochDuties.CurrentEpochDuties,
 	}, nil
 }
 
 func (c *beaconApiValidatorClient) dutiesForEpoch(
 	ctx context.Context,
+	dutiesContainer *ethpb.ValidatorDutiesContainer,
 	epoch primitives.Epoch,
 	vals []validatorForDuty,
 	fetchSyncDuties bool,
-) ([]*ethpb.ValidatorDuty, error) {
+) error {
```

Comment:
why did we change the method signature ?

Reply:
Because we need the dependent roots that are returned from the duties endpoints.

Reply:
makes sense since we need the dependent root now along with the duties